### PR TITLE
Improve coverage 10

### PR DESCRIPTION
**Pull request by tsmith023**

### CODE REVIEW

File: weaviate/collections/classes/config_methods.py [hunk 1]
```diff
@@ -181,7 +181,9 @@ def _collection_config_from_json(schema: Dict[str, Any]) -> _CollectionConfig:
             key=schema["shardingConfig"]["key"],
             strategy=schema["shardingConfig"]["strategy"],
             function=schema["shardingConfig"]["function"],
-        ),
+        )
+        if not schema["multiTenancyConfig"]["enabled"]
+        else None,
```
Comment (dirkkul) [inline weaviate/collections/classes/config_methods.py:186]: what is this change for? You can have multiple shards without MT

File: integration/test_collection_config.py [hunk 2]
```diff
@@ -650,22 +650,41 @@ def test_config_export_and_recreate_from_config(collection_factory: CollectionFa
...
 
     name = "TestCollectionConfigExportAndRecreateFromDict"
     conf.name = name
     dconf = conf.to_dict()
 
-    client = weaviate.connect_to_local()
-    client.collections.create_from_dict(dconf)
-    assert conf == client.collections.get(name).config.get()
-    client.collections.delete(name)
+    try:
```
Comment (dirkkul) [inline integration/test_collection_config.py:677]: why this?

File: integration/test_collection_config.py [hunk 3]
```diff
@@ -650,22 +650,41 @@ def test_config_export_and_recreate_from_config(collection_factory: CollectionFa
 
 def test_config_export_and_recreate_from_dict(collection_factory: CollectionFactory) -> None:
```
Comment (dirkkul) [inline integration/test_collection_config.py:651]: we have a similar test in test_client - but we can leave it for now

File: weaviate/collections/classes/config_methods.py (hunk 1 above)
Comment (tsmith023) [reply weaviate/collections/classes/config_methods.py:186]: You can't have sharding and multi-tenancy though, which is what this is doing. I will reverse the logic to make it clearer

File: integration/test_collection_config.py (hunk 2 above)
Comment (tsmith023) [reply integration/test_collection_config.py:677]: Don't really need it, will remove (just cleans up resources)

File: integration/test_collection_config.py (hunk 3 above)
Comment (tsmith023) [reply integration/test_collection_config.py:651]: just noticed, the one in this PR tests much more so let's keep it but we should definitely merge them

**Review by dirkkul — APPROVED** [review]

**Pull request merged**